gitdir/gitdir.py

```diff
@@ -30,31 +44,107 @@ def print_text(text, color="default", in_place=False, **kwargs):  # type: (str,
     print(COLOR_NAME_TO_CODE[color] + text + Style.RESET_ALL, **kwargs)
 
 
+def prompt_yes_no(question: str, default: bool = True) -> bool:
+    """
+    Prompt user for a yes/no question.
+
+    :param question: question to ask
+    :param default: default answer if user just presses enter
+    :return: True if user answers yes, False if user answers no
+    """
+    if default:
+        yes_no = "Y/n"
+    else:
+        yes_no = "y/N"
+
+    while True:
+        print_text("{} [{}] ".format(question, yes_no), end="")
+        choice = input().lower()
+        if choice in {"y", "yes"}:
+            return True
+        elif choice in {"n", "no"}:
+            return False
+        elif choice == "":
+            return default
+        else:
+            print_text("Please respond with 'yes' or 'no' (or 'y' or 'n').")
+
+
 def create_url(url):
     """
     From the given url, produce a URL that is compatible with Github's REST API. Can handle blob or tree paths.
     """
-    repo_only_url = re.compile(r"https:\/\/github\.com\/[a-z\d](?:[a-z\d]|-(?=[a-z\d])){0,38}\/[a-zA-Z0-9]+$")
+    repo_only_url = re.compile(
+        r"https:\/\/github\.com\/[a-z\d](?:[a-z\d]|-(?=[a-z\d])){0,38}\/[a-zA-Z0-9]+$"
+    )
     re_branch = re.compile("/(tree|blob)/(.+?)/")
 
     # Check if the given url is a url to a GitHub repo. If it is, tell the
     # user to use 'git clone' to download it
-    if re.match(repo_only_url,url):
-        print_text("✘ The given url is a complete repository. Use 'git clone' to download the repository",
-                   "red", in_place=True)
+    if re.match(repo_only_url, url):
+        print_text(
+            "✘ The given url is a complete repository. Use 'git clone' to download the repository",
+            "red",
+            in_place=True,
+        )
         sys.exit()
 
     # extract the branch name from the given url (e.g master)
     branch = re_branch.search(url)
-    download_dirs = url[branch.end():]
-    api_url = (url[:branch.start()].replace("github.com", "api.github.com/repos", 1) +
-              "/contents/" + download_dirs + "?ref=" + branch.group(2))
+    if branch is None:
+        print_text(
+            "✘ Could not find branch name in the given url", "red", in_place=True
```

Comment:
Optimally, this would try common branch names like `main` and `master`, but that's out of scope for this PR